docs/en/reference/working-with-associations.rst

```diff
@@ -559,6 +559,13 @@ OrphanRemoval works with one-to-one, one-to-many and many-to-many associations.
     If you neglect this assumption your entities will get deleted by Doctrine even if
     you assigned the orphaned entity to another one.
 
+.. note::
+
+    ``orphanRemoval=true`` option should be used in combination with `cascade=["persist"]` option
+    as the child entity that is manually persisted will not be deleted automatically by Doctrine
+    when collection is still instance of ArrayCollection (before first flush / hydration).
+    This is a Doctrine limitation since ArrayCollection does not have access to UnitOfWork.
```

Comment:
```suggestion
    ``orphanRemoval=true`` option should be used in combination with `cascade=["persist"]` option
    as the child entity, that is manually persisted, will not be deleted automatically by Doctrine
    when a collection is still an instance of ArrayCollection (before first flush / hydration).
    This is a Doctrine limitation since ArrayCollection does not have access to a UnitOfWork.
```